spec/features/users/sign_in_spec.rb

```diff
@@ -425,6 +425,17 @@
     end
   end
 
+  context 'user attempts sign in with bad personal key' do
+    it 'remains on the login with personal key page' do
+      user = create(:user, :signed_up)
+      signin(user.email, user.password)
+      choose_another_security_option('personal_key')
+      enter_personal_key(personal_key: 'foo')
+      click_submit_default
+      expect(page).to have_current_path(login_two_factor_personal_key_path)
```

Comment:
can we also add a test for the error message?

Reply:
done